File: src/main/java/project/rpg/skill/SkillDic.java
package project.rpg.skill;

import org.bukkit.entity.Player;
import project.rpg.skill.base.SkillBase;
import project.rpg.skill.magic.fire.MeteorStrike;
import project.rpg.skill.test.ShoonBow;
import project.rpg.skill.test.TpArrow;

import java.util.ArrayList;

public class SkillDic { //이거 SkillCommand 에서 switch 문 말고 if로 단순화 할려고

    private static final ArrayList<String> _skillList = new ArrayList<>();  //전체 스킬 이름 리스트

    public static SkillBase makeSkill(Player player, String skillName) {
        if(skillName.equals(SkillType.METEOR_STRIKE.getSkillName())) {
            return new MeteorStrike(player);
        } else if (skillName.equals(SkillType.TP_ARROW.getSkillName())) {
            return new TpArrow(player);
        } else if (skillName.equals(SkillType.SHOONBOW.getSkillName())) {
            return new ShoonBow(player);
        }
        return null;
    }

    public static void addAll() {
        _skillList.add(SkillType.METEOR_STRIKE.getSkillName());
        _skillList.add(SkillType.TP_ARROW.getSkillName());
        _skillList.add(SkillType.SHOONBOW.getSkillName());
    }

    public static boolean isExist(String skillName) {
        return _skillList.contains(skillName);
    }
}

File: src/main/java/project/rpg/skill/SkillDic.kt
package project.rpg.skill

import org.bukkit.entity.Player
import project.rpg.skill.base.SkillBase
import project.rpg.skill.magic.earth.Binding
import project.rpg.skill.magic.electricity.*
import project.rpg.skill.magic.fire.*
import project.rpg.skill.magic.water.*
import project.rpg.skill.magic.wind.HeavenWing
import project.rpg.skill.magic.wind.Tempest
import project.rpg.skill.tmp.ShoonBow
import project.rpg.skill.tmp.TpArrow

object SkillDic {
    //이거 SkillCommand 에서 switch 문 말고 if로 단순화 할려고
    val skillList = ArrayList<String>() //전체 스킬 이름 리스트
    fun makeSkill(player: Player?, skillName: String): SkillBase? {
        return when (skillName) {
            SkillType.METEOR_STRIKE.skillName -> {
                MeteorStrike()
            }
            SkillType.FLAME_BURST.skillName -> {
                FlameBurst()
            }
            SkillType.TP_ARROW.skillName -> {
                TpArrow()
            }
            SkillType.SHOONBOW.skillName -> {
                ShoonBow(player)
            }
            SkillType.EXPLOSION.skillName -> {
                Explosion()
            }
            SkillType.LAVA_ZONE.skillName -> {
                LavaZone()
            }
            SkillType.BLAZING_MARK.skillName -> {
                BlazingMark()
            }
            SkillType.INFERNO.skillName -> {
                Inferno()
            }
            SkillType.FLARE_CLOCK.skillName -> {
                FlareClock()
            }
            SkillType.PURE_SHIELD.skillName -> {
                PureShield()
            }
            SkillType.WATER_ARROW.skillName -> {
                WaterArrow()
            }
            SkillType.FROZEN_TRACE.skillName -> {
                FrozenTrace()
            }
            SkillType.ICE_SPEAR.skillName -> {
                IceSpear()
            }
            SkillType.ETERNAL_FROST.skillName -> {
                EternalFrost()
            }
            SkillType.LIGHTNING_STORM.skillName -> {
                LightningStorm()
            }
            SkillType.THUNDER_CHARGING.skillName -> {
                ThunderCharging()
            }
            SkillType.LIGHTNING_CHAIN.skillName -> {
                LightningChain()
            }
            SkillType.SHOCK_WAVE.skillName -> {
                ShockWave()
            }
            SkillType.GIGANTIC_THUNDER.skillName -> {
                GiganticThunder()
            }
            SkillType.HEAVEN_WING.skillName -> {
                HeavenWing()
            }
            SkillType.TEMPEST.skillName -> {
                Tempest()
            }
            SkillType.BINDING.skillName -> {
                Binding()
            }
            else -> null
        }
    }

    fun addAll() {
        for (skill in SkillType.values()) {
            skillList.add(skill.skillName)
        }
    }

    fun isExist(skillName: String): Boolean {
        return skillList.contains(skillName)
    }
}